### PR TITLE
Adding Docker Setup

### LINKED ISSUE #378: Feat: Dockerising the codebase.
#### Reason/Context

Dokerising the codebase will help contributors to setup the codebase easily , saving their time and machine resources 

#### Description

What changes have to be 

Adding docker setup to the codebase.

Tasks to be done 

- [ ] Adding docker setup by their native commands 
- [ ] Creating make commands
- [ ]  Updating the installation readme file 

**Do you think this work might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

No

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
**GitHub Issue:** #378 
closes #378 

**Summary**:

1. Dockerised the setup by adding docker file for creating the images and run those images in docker containers.
2. Added make commands to shorten those commands and can be used easily by developers.

**files added**
1. dockerfile
2. dockerignorefile
3. makefile

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

Yes

### CODE REVIEW

**Comment by DhairyaMajmudar** [comment]
@benjagm I have added the docker setup for codebase.

You can check the work by following 
1. Running `make install` command for creating docker images.
2. Running `make run` command for running the website in `localhost:3000`

**Review by benjagm — APPROVED** [review]
Can we modify the readme to tell the users that **they have the option** to use docker as well?

**Review by benjagm — CHANGES_REQUESTED** [review]
Can we modify the readme to explain users that they have the option of using dokcer if they prefer it?

**Comment by DhairyaMajmudar** [comment]
@benjagm I have added the docker installation guide in the readme pls. have a look : )

File: README.md [hunk 1]
```diff
@@ -61,6 +61,27 @@ Build static files on /out folder
 yarn build
 ```
 
+### Run locally using Docker
```
Comment (benjagm) [inline README.md:64]:
```suggestion
### Run locally using Docker

If you are a Docker lover, you have the option to use it following these instructions.
```

**Review by benjagm — CHANGES_REQUESTED** [review]
Looks good. Just left the last suggestion to make it clear that docker is just another option to run it.

**Comment by DhairyaMajmudar** [comment]
Thanks you suggestion , I have committed it.

**Review by benjagm — APPROVED** [review]

**Comment by benjagm** [comment]
Good job here!

**Pull request merged**